src/main/java/net/minecraftforge/common/capabilities/IComparableCapabilityProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.capabilities;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Defines a Capability Provider that can be checked for equivalence with another provider.
+ */
+public interface IComparableCapabilityProvider<B extends IComparableCapabilityProvider<B>> extends ICapabilityProvider
```

Comment:
These methods could maybe be made protected methods in CapabilityProvider that ItemStack overrides to make public, to reduce class and interface spamming.

Reply:
They can just go onto IForgeItemStack, tbh